### PR TITLE
attestion: make `InvalidAttestationError` non-fatal in CI

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

I don't think I've seen an `InvalidAttestationError` that wasn't some
sort of network problem (e.g., rate limit, connection timeout, 503).
Let's emit a warning instead of erroring out.

Note that `MissingAttestationError` is still fatal, and that will still
produce errors in CI.

### CODE REVIEW

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
I appreciate the intent here but 👎🏻 as-is.

If we're going to make this non-fatal it feels like it defeats the entire point of having attestation. 

I'd like to have us try a couple of retries with an exponential backoff before we do this.

**Comment by dduugg** [comment]
> ```
> attestation.rb:135: Method `system_command!` does not exist on `T.class_of(Homebrew::Attestation)` https://srb.help/7003
>      135 |        result = system_command!(gh_executable, args: cmd,
>                            ^^^^^^^^^^^^^^^
> Errors: 1
> ```
> 
> Huh, interesting. Not even code I touched though. Would love a suggestion here. CC @dduugg ?

Try pairing this with an explicit `begin`, i don't think the way code aligns matches the control flow. (Then, I think you should be getting a different type error, since you've made the return type nilable)

**Comment by carlocab** [comment]
Implemented retry logic instead.

File: Library/Homebrew/attestation.rb [hunk 1]
```diff
@@ -246,6 +248,15 @@ def self.check_core_attestation(bottle)
       end
 
       backfill_attestation
+    rescue InvalidAttestationError
+      @attestation_retry_count ||= T.let(Hash.new(0), T.nilable(T::Hash[Bottle, Integer]))
+      raise if @attestation_retry_count[bottle] >= ATTESTATION_MAX_RETRIES
+
+      sleep_time = 3 ** @attestation_retry_count[bottle]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/attestation.rb:255]:
```suggestion
      sleep_time = 2 ** @attestation_retry_count[bottle]
```
would be fine with me if you want this to take smaller jumps/be quicker to fail

File: Library/Homebrew/attestation.rb [hunk 2]
```diff
@@ -179,6 +179,8 @@ def self.check_attestation(bottle, signing_repo, signing_workflow = nil, subject
       attestation
     end
 
+    ATTESTATION_MAX_RETRIES = 5
```
Comment (MikeMcQuaid) [inline Library/Homebrew/attestation.rb:182]:
```suggestion
    ATTESTATION_MAX_RETRIES = 2
```
or 
```suggestion
    ATTESTATION_MAX_RETRIES = 3
```
would be fine with me if you want to fail earlier

**Review by MikeMcQuaid — APPROVED** [review]
Thanks @carlocab! Have suggested speeding up failure time so we can evaluate if this is something we should back out entirely for now. Feel free to merge as-is if preferred.

**Comment by carlocab** [comment]
Let's keep the retry timing as is. The case for backing this out if we need to will be stronger if we were more patient about retries.

**Pull request merged**